processes/consumer/flush.go

```diff
@@ -56,64 +57,76 @@ func Flush(ctx context.Context, inMemDB *models.DatabaseData, dest destination.B
 				return
 			}
 
-			// Lock the tables when executing merge / append.
 			_tableData.Lock()
 			defer _tableData.Unlock()
 			if _tableData.Empty() {
 				return
 			}
 
-			// This is added so that we have a new temporary table suffix for each merge / append.
-			_tableData.ResetTempTableSuffix()
-
 			start := time.Now()
 			tags := map[string]string{
-				"what":     "success",
 				"mode":     _tableData.Mode().String(),
 				"table":    _tableName,
 				"database": _tableData.TopicConfig().Database,
 				"schema":   _tableData.TopicConfig().Schema,
 				"reason":   args.Reason,
 			}
 
-			var err error
-			action := "merge"
-			// Merge or Append depending on the mode.
-			if _tableData.Mode() == config.History {
-				err = dest.Append(_tableData.TableData, false)
-				action = "append"
-			} else {
-				err = dest.Merge(_tableData.TableData)
-			}
+			what, err := retry.WithRetriesAndResult(retryCfg, func(_ int, _ error) (string, error) {
+				return flush(ctx, dest, _tableName, _tableData)
+			})
 
 			if err != nil {
-				tags["what"] = "merge_fail"
-				tags["retryable"] = fmt.Sprint(dest.IsRetryableError(err))
-				slog.With(logFields...).Error(fmt.Sprintf("Failed to execute %s, not going to flush memory, will sleep for 3 seconds before continuing...", action), slog.Any("err", err))
-				time.Sleep(3 * time.Second)
+				slog.Error(fmt.Sprintf("Failed to %s", action), slog.Any("err", err), slog.String("tableName", _tableName))
 			} else {
-				slog.Info(fmt.Sprintf("%s success, clearing memory...", stringutil.CapitalizeFirstLetter(action)), logFields...)
-				commitErr := commitOffset(ctx, _tableData.TopicConfig().Topic, _tableData.PartitionsToLastMessage)
-				if commitErr == nil {
-					inMemDB.ClearTableConfig(_tableName)
-				} else {
-					tags["what"] = "commit_fail"
-					slog.Warn("Commit error...", slog.Any("err", commitErr))
-				}
+				inMemDB.ClearTableConfig(_tableName)
+				slog.Info(fmt.Sprintf("%s success, clearing memory...", stringutil.CapitalizeFirstLetter(action)), slog.String("tableName", _tableName))
 			}
+
+			tags["what"] = what
 			metricsClient.Timing("flush", time.Since(start), tags)
 		}(tableName, tableData)
 	}
-	wg.Wait()
 
+	wg.Wait()
 	return nil
 }
+
+func flush(ctx context.Context, dest destination.Baseline, _tableName string, _tableData *models.TableData) (string, error) {
+	// This is added so that we have a new temporary table suffix for each merge / append.
+	_tableData.ResetTempTableSuffix()
+
+	// Merge or Append depending on the mode.
+	var err error
+	if _tableData.Mode() == config.History {
+		err = dest.Append(_tableData.TableData, false)
+	} else {
+		err = dest.Merge(_tableData.TableData)
+	}
+
+	if err != nil {
+		return "merge_fail", fmt.Errorf("failed to flush: %w", err)
+	}
+
+	if err = commitOffset(ctx, _tableData.TopicConfig().Topic, _tableData.PartitionsToLastMessage); err != nil {
+		return "commit_fail", fmt.Errorf("failed to commit kafka offset: %w", err)
```

Comment:
If Kafka commit fails, this will now mark the flush as errored `what = "commit_fail"` which would prevent `inMemDB.ClearTableConfig(_tableName)`